### ISSUE TITLE
ReadMe for getting this project working?

### ISSUE THREAD

**Issue by sirwesley** [Aldaviva/BehringerXTouchExtender#1]
Trying to get this project compiling and running.  Do you have any hints to get me started?  

What does this project do?  I've got an X Touch Extender in MIDI mode.  Trying to write some text to the Scribble Strips and your code looks to be the closest solution to getting that done.   

What version of your KoKo library should I be using?  

Programming experience isn't so great with C#.  More a Typescript/Javascript dev.  

I've opened the solution file with VS2022 and attempted to build, but the 'sample' program complains....

```
fader.DesiredPosition.Connect(trackId / (device.TrackCount - 1.0));
device.GetScribbleStrip(trackId).TopText.Connect($"Track {trackId + 1}");
device.GetScribbleStrip(trackId).BottomText.Connect(new string('.', trackId));
device.GetScribbleStrip(trackId).TopTextColor.Connect(ScribbleStripTextColor.Dark);
device.GetScribbleStrip(trackId).BottomTextColor.Connect(ScribbleStripTextColor.Light);
device.GetScribbleStrip(trackId).BackgroundColor
        .Connect(trackId == 0 ? ScribbleStripBackgroundColor.White : (ScribbleStripBackgroundColor) trackId); //avoid black background because it makes text illegible

```
        
```

1>------ Build started: Project: Sample, Configuration: Debug Any CPU ------
1>E:\SirWesley\Development\XTouchCS\BehringerXTouchExtender\Sample\Sample.csproj : warning NU1701: Package 'CSCore 1.2.0' was restored using '.NETFramework,Version=v4.6.1, .NETFramework,Version=v4.6.2, .NETFramework,Version=v4.7, .NETFramework,Version=v4.7.1, .NETFramework,Version=v4.7.2, .NETFramework,Version=v4.8, .NETFramework,Version=v4.8.1' instead of the project target framework 'net6.0'. This package may not be fully compatible with your project.
1>E:\SirWesley\Development\XTouchCS\BehringerXTouchExtender\Sample\Sample.cs(79,35,79,79): error CS1503: Argument 1: cannot convert from 'double' to 'KoKo.Property.Property<double>'
1>E:\SirWesley\Development\XTouchCS\BehringerXTouchExtender\Sample\Sample.cs(81,54,81,76): error CS1503: Argument 1: cannot convert from 'string' to 'KoKo.Property.Property<string>'
1>E:\SirWesley\Development\XTouchCS\BehringerXTouchExtender\Sample\Sample.cs(82,57,82,81): error CS1503: Argument 1: cannot convert from 'string' to 'KoKo.Property.Property<string>'
1>E:\SirWesley\Development\XTouchCS\BehringerXTouchExtender\Sample\Sample.cs(83,59,83,86): error CS1503: Argument 1: cannot convert from 'BehringerXTouchExtender.ScribbleStripTextColor' to 'KoKo.Property.Property<BehringerXTouchExtender.ScribbleStripTextColor>'
1>E:\SirWesley\Development\XTouchCS\BehringerXTouchExtender\Sample\Sample.cs(84,62,84,90): error CS1503: Argument 1: cannot convert from 'BehringerXTouchExtender.ScribbleStripTextColor' to 'KoKo.Property.Property<BehringerXTouchExtender.ScribbleStripTextColor>'
1>E:\SirWesley\Development\XTouchCS\BehringerXTouchExtender\Sample\Sample.cs(86,18,86,108): error CS1503: Argument 1: cannot convert from 'BehringerXTouchExtender.ScribbleStripBackgroundColor' to 'KoKo.Property.Property<BehringerXTouchExtender.ScribbleStripBackgroundColor>'
1>Done building project "Sample.csproj" -- FAILED.
========== Build: 0 succeeded, 1 failed, 1 up-to-date, 0 skipped ==========
```

**Comment by sirwesley** [comment]
Actually, I just succeeded in running this app by building the solution with those lines commented out.  I do see that the X-Touch Extender connected successfully and is outputting data based on the buttons I press.  Also fun to see the VU Meters dancing to music.  So, now just figuring out how to fix the scribble strips???

**Comment by Aldaviva** [comment]
Oh sorry I totally thought I had already written a readme for this. I'll add that.

So far I haven't packed this project to make it easy to consume with a dependency manager like NuGet, rather than manually managing and building it yourself. I'll try to work on this too. I was writing this to use in a downstream project, but that ran into a roadblock so I paused development.

In the meantime, I think you may be able to depend on the [latest version of KoKo on NuGet](https://www.nuget.org/packages/KoKo/) using the dotnet, Install-Package, or PackageReference commands at the top of that page.

I'll let you know when there's a real package to use.

**Comment by Aldaviva** [comment]
Now that I think about it, I believe the sample scribble strip code uses a new shorthand syntax for using constants with connectableproperties, and it may not exist in the latest stable version of koko yet. I'll try to cut a release of that too. In the meantime, you can probably replace the `.Connect(foo)` calls with `.Connect(new StoredProperty<FooType>(foo))`.

**Comment by Aldaviva** [comment]
I released [KoKo 2.2.0 on NuGet](https://www.nuget.org/packages/KoKo/2.2.0) with the new `ConnectableProperty<T>.Connect(T)` method overload.

You can now remove the `<Reference Include="KoKo">` element from your `.csproj` file and replace it with the official NuGet build using one of the following techniques.
- searching for KoKo in Visual Studio (Project › Manage NuGet Packages… › Browse) and clicking Install
- pasting `<PackageReference Include="KoKo" Version="2.2.0" />` into an `<ItemGroup>` element in the `.csproj` file
- running `dotnet add package KoKo --version 2.2.0` in the project directory
- running `NuGet\Install-Package KoKo -Version 2.2.0` in Visual Studio's Package Manager Console window

I will do the same in my upstream repository and push it when I can.

This should fix the CS1503 compilation issues. The readme still needs to be written.

---

Now you should be able to set text and colors for scribble strips using the following syntax.
```cs
device.GetScribbleStrip(0).TopText.Connect("Hello");
```

For a complete example that changes all text and color properties of a scribble strip, refer to the [Library Usage example in the Scribble Strips wiki page](https://github.com/Aldaviva/BehringerXTouchExtender/wiki/Scribble-strips#library-usage).

**Comment by sirwesley** [comment]
Awesome!  Works perfectly now.  Thanks!
